### DISCUSSION TITLE
Prop to disable default selection of first MenuItem when opening a Menu

### DISCUSSION THREAD

**Discussion by vinodkola1337** [primefaces/community#503] [PrimeVue]
When we open a Menu, the first Menuitem gets selected by default. I know this is by [design](https://github.com/primefaces/primevue/issues/3162), but I am curious to understand the reason. Is it for keyboard interactions?

I also noticed that there is a prop called 'autoOptionFocus' for Dropdown component for disabling this behavior. Is it possible to have a similar prop for the other Menu components too? It would be a great addition for us.

**Comment by tugcekucukoglu** [comment]
We need more community demand on this issue.

**Comment by samuelwei** [comment]
I think this is related to the issue: https://github.com/primefaces/primevue/issues/4481 ?

**Reply by samuelwei** [reply]
I'm also looking for an option to disable this, or even better be able to programmatically set the focus to an item to show the current state. Use case: Use Menu with popover as a dropdown menu button for a locale switcher, the focus should always be on the current locale.

**Comment by Caarnus** [comment]
I would also like to request this feature. It makes the styling look odd by default and while there are work arounds it seems like something that should be configurable as part of the component.

**Comment by VororRich** [comment]
Another issue this presents is if you are using a menu to toggle an option (such as a theme menu).  If you use class to background highlight the currently selected option, the focus hides it.

**Comment by MAN-Sendance** [comment]
I support the idea of getting an option for removing the focus.

**Comment by AniMZ711** [comment]
I’m experiencing the same issue, and it’s a significant problem when developing accessible single-page websites. It looks very odd when navigating to a section, only for the focus to unexpectedly jump to the first item. Is there any update or a current workaround available?
